src/duckling/game/collision/edit-collision-types.component.ts

```diff
@@ -0,0 +1,72 @@
+import {Component, ViewContainerRef} from '@angular/core';
+import {MdDialogRef} from '@angular/material';
+import {Observable} from 'rxjs';
+
+import {openDialog} from '../../util/md-dialog';
+import {StoreService} from '../../state/store.service';
+import {ProjectService} from '../../project/project.service';
+
+import {CollisionTypesService} from './collision-types.service';
+
+/**
+ * Dialog to allow the user to manager their collision types
+ */
+@Component({
+    selector: "dk-edit-collision-types",
+    styleUrls: ["./duckling/layout.css", "./duckling/game/collision/edit-collision-types.component.css"],
+    template: `
+        <dk-section headerText="Create Collision Type">
+            <div>
+                <dk-input class="dk-inline"
```

Comment:
This pattern is being used in the map select, the entity rename tool, and here. That means it should probably be extracted into a generic control.